src/models/karaoke.py

```diff
@@ -0,0 +1,220 @@
+from dataclasses import dataclass
+from datetime import timedelta
+from functools import reduce
+from itertools import accumulate
+from typing import Sequence
+
+from src.models import SongLine
+
+from src.fonts import FontScaler
+
+import pyass
+from pyass import Style
+
+
+class KLine:
+    pass
+
+
+class KSyl:
+    pass
+
+
+@dataclass
+class KChar:
+    char: str
+    i: int
+    sylI: int
+    fadeOffset: timedelta() = timedelta()
+    karaStart: timedelta() = timedelta()
+    karaEnd: timedelta() = timedelta()
+    karaDuration: timedelta() = timedelta()
```

Comment:
Can this be a property calculated from `karaStart` and `karaEnd`?

Reply:
It can, I just copied over some of the data from https://aegisite.vercel.app/docs/latest/automation/lua/modules/karaskel.lua/ directly so I'll change it